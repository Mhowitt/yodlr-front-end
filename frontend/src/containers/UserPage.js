import React, { Component } from "react";
import axios from "axios";

export default class UserPage extends Component {
  state = {
    firstName: "",
    lastName: "",
    email: "",
    hasRegistered: false
  };
  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };
  handleSubmit = e => {
    let { firstName, lastName, email } = this.state;
    e.preventDefault();
    axios
      .post("http://localhost:3000/users", {
        firstName,
        lastName,
        email
      })
      .then(() => {
        this.setState({
          hasRegistered: true
        });
      })
      .catch(() => {
        alert("something went wrong!");
      });
  };
  render() {
    if (!this.state.hasRegistered) {
      return (
        <div>
          <h1>Welcome!</h1>
          <form onSubmit={this.handleSubmit}>
            <div>
              <label htmlFor="firstName">First Name:</label>
              <input
                type="text"
                name="firstName"
                id="firstName"
                value={this.state.firstName}
                onChange={this.handleChange}
              />
            </div>
            <div>
              <label htmlFor="lastName">Last Name:</label>
              <input
                type="text"
                name="lastName"
                id="lastName"
                value={this.state.lastName}
                onChange={this.handleChange}
              />
            </div>
            <div>
              <label htmlFor="email">Email:</label>
              <input
                type="email"
                name="email"
                id="email"
                onChange={this.handleChange}
                value={this.state.email}
              />
            </div>
            <button>Register!</button>
          </form>
        </div>
      );
    } else {
      return <h2>Thank you for signing up!</h2>;
    }
  }
}
