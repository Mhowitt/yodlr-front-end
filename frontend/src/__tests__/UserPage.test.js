import React from "react";
import UserPage from "../containers/UserPage";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<UserPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
