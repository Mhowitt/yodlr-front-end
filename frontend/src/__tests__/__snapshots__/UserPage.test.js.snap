// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Welcome!
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="firstName"
      >
        First Name:
      </label>
      <input
        id="firstName"
        name="firstName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="lastName"
      >
        Last Name:
      </label>
      <input
        id="lastName"
        name="lastName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="email"
      >
        Email:
      </label>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <button>
      Register!
    </button>
  </form>
</div>
`;
